### PR TITLE
2086: swap new submission wizard

### LINKED ISSUE #2086: Change the code in SubmissionWizard to be that found in NewSubmissionWizard
## Task

We need to remove the `newSubmit` route and implement the code found in NewSubmissionWizard on the `submit` route. Copy code over to the SubmissionWizard file to maintain history.

### PR DESCRIPTION
**Pull request by erezmus**
* copied NewSubmissionWizard.js into SubmissionWizard.js
* started replacing tests
* installed @testing-library/react as enzyme doesn't work with hooks yet

#### Background

Turns on the new submission wizard

#### Any relevant tickets

Closes #2086, #2119, #2120

#### How has this been tested?
**Reviewers** ensure that these test requirements are also reviewed.

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. 

- [ ] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

#### UI Changes

- [ ] Has been reviewed against Designs
- [ ] Necessary updates to styleguide

### CODE REVIEW

**Comment by diversemix** [comment]
@erezmus ... I've merged in develop as there were changes to the SubmissionWizard from the changes to use `suggestions`

**Comment by diversemix** [comment]
Apparently this is still awaiting tests... if anyone can add more info

File: packages/component-submission/client/pages/SubmissionWizard.test.js [hunk 1]
```diff
@@ -1,109 +1,186 @@
...
+
+    const makeProps = path => ({
+      data: { manuscript: {} },
+      match: { path: '/submit/id', url: '', params: { id: 'id' } },
+      history: { location: { pathname: path }, push: pushHistory },
+      updateManuscript: jest.fn(),
+      submitManuscript: jest.fn(),
+    })
+
+    const setup = (historyLocation = []) => ({ children }) => (
```
Comment (Cooryd) [inline packages/component-submission/client/pages/SubmissionWizard.test.js]: I'll work on a change to have this defined once out of the test case scope

File: packages/component-submission/client/components/WizardSubmit.js [hunk 2]
```diff
@@ -15,7 +15,7 @@ const WizardSubmit = ({
     {({ showModal, hideModal, isModalVisible }) => (
       <React.Fragment>
         <Button
-          data-test-id="submit"
+          data-testid="submit"
```
Comment (petereast) [inline packages/component-submission/client/components/WizardSubmit.js]:
```suggestion
          data-testid="submit"
```
```suggestion
          data-test-id="submit"
```
Is this deliberate?

File: packages/component-submission/client/pages/SubmissionWizard.test.js [hunk 3]
```diff
@@ -1,109 +1,269 @@
...
+    expect(
+      renderWithPath('/submit/id/disclosure').getAllByText(
+        'DisclosureStepPage',
+      ),
+    ).toBeTruthy()
+  })
 
-    thing.instance().setIsUploading(true)
+  it('should change the location on next', async () => {
+    let opts = null
```
Comment (petereast) [inline packages/component-submission/client/pages/SubmissionWizard.test.js]: Why not use a new variable each time?

File: packages/component-submission/client/pages/SubmissionWizard.test.js [hunk 4]
```diff
@@ -1,109 +1,269 @@
...
-    const mockSubmit = jest.fn(() =>
-      new Promise(resolve => {
-        expect(thing.instance().state.suspendSave).toBe(true)
-        resolve()
-      }).catch(err => {
-        // eslint-disable-next-line no-undef
-        fail(err)
-      }),
+  it('should change the location on back except on first step', async () => {
+    let opts = null
```
Comment (petereast) [inline packages/component-submission/client/pages/SubmissionWizard.test.js]: Again, why reuse the same variable? I'll be much more readable if it's a more descriptive name per test case

File: packages/component-submission/client/components/WizardSubmit.js (hunk 2 above)
Comment (erezmus) [reply packages/component-submission/client/components/WizardSubmit.js]: Yes as the standard is data-testid which is used by getTestId

Comment (petereast) [reply packages/component-submission/client/components/WizardSubmit.js]: Ah, right

Comment (petereast) [reply packages/component-submission/client/components/WizardSubmit.js]: Could this break the browser tests?

Comment (erezmus) [reply packages/component-submission/client/components/WizardSubmit.js]: so actually, we use data-test-id throughout the browser tests, so lets change those back to data-test-id and configure it to use that

File: packages/component-submission/client/pages/SubmissionWizard.test.js [hunk 5]
```diff
@@ -1,109 +1,150 @@
...
-    expect(mockSubmit).toHaveBeenCalledTimes(1)
+    const SUPPRESSED_PREFIXES = [
+      'Warning: Do not await the result of calling ReactTestUtils.act(...)',
+      'Warning: An update to %s inside a test was not wrapped in act(...)',
+    ]
+
+    beforeAll(() => {
+      // disable formik warnings
+      // see https://stackoverflow.com/questions/55181009/jest-react-testing-library-warning-update-was-not-wrapped-in-act
+      // https://github.com/facebook/react/pull/14853 removes the need for this once we upgrade to react-dom 16.9
```
Comment (Cooryd) [inline packages/component-submission/client/pages/SubmissionWizard.test.js:116]: raise issue for this

File: packages/component-submission/client/components/WizardSubmit.js [hunk 6]
```diff
@@ -15,7 +15,7 @@ const WizardSubmit = ({
     {({ showModal, hideModal, isModalVisible }) => (
       <React.Fragment>
```
Comment (Cooryd) [inline packages/component-submission/client/components/WizardSubmit.js]: Raise issue to destruct React import to use `Fragment`

**Review by diversemix — APPROVED** [review]

**Review by will-byrne — APPROVED** [review]

**Review by petereast — APPROVED** [review]

**Pull request merged**